services/web/server/src/simcore_service_webserver/projects/_controller/projects_rest_schemas.py

```diff
@@ -167,6 +168,22 @@ def _search_check_empty_string(cls, v):
             return None
         return v
 
+    _template_type_null_or_none_str_to_none_validator = field_validator(
+        "template_type", mode="before"
+    )(null_or_none_str_to_none_validator)
+
+    @model_validator(mode="after")
+    def check_template_type_compatibility(self):
+        if (
+            self.project_type in [ProjectTypeAPI.all, ProjectTypeAPI.user]
+            and self.template_type is not None
+        ):
+            msg = (
+                "When project type is `all` or `user` the template_type should be None"
```

Comment:
I find these more easy to debug when you change the code. Simpler to fix as well
```suggestion
                f"When {self.project_type=} is `all` or `user` the {self.template_type=} should be None"
```